### PR TITLE
🐛 Destination Clickhouse-strict-encrypt: Remove SSL toggle in spec and add SSL modes

### PR DESCRIPTION
**Pull request by StellaMaUCI**
## What
We are fixing the issue below: 
https://github.com/airbytehq/airbyte/issues/21262
The SSL toggle is redundant when the SSL modes work. 

## How
Remove the ssl key from spec.json for destination-Clickhouse and rely on ssl_mode key as the source of truth.

### CODE REVIEW

**Comment by StevenReitsma** [comment]
Nice! However, I don't think Clickhouse supports things like "verify" or "prefer". It's either "none" or "strict", according to the documentation. I've previously added the proper SSL mode switch to the (unmerged) PR #25914.

**Comment by StellaMaUCI** [comment]
> Nice! However, I don't think Clickhouse supports things like "verify" or "prefer". It's either "none" or "strict", according to the documentation. I've previously added the proper SSL mode switch to the (unmerged) PR #25914.

Many thanks for your warm reminder! We will refer to your PR and make the necessary fixes to ours accordingly.

File: airbyte-config-oss/init-oss/src/main/resources/seed/destination_definitions.yaml
Comment (sajarin) [inline airbyte-config-oss/init-oss/src/main/resources/seed/destination_definitions.yaml:1]: Can you remove this file from the branch? This has been file has been deleted from master.

File: airbyte-config-oss/init-oss/src/main/resources/seed/destination_specs.yaml
Comment (sajarin) [inline airbyte-config-oss/init-oss/src/main/resources/seed/destination_specs.yaml:1]: Can you remove this file from the branch? This has been file has been deleted from master.

File: airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json
Comment (sajarin) [inline airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json:1]: Can you remove this file from the branch? This has been file has been deleted from master.

File: airbyte-integrations/connectors/destination-clickhouse-strict-encrypt/src/test/resources/expected_spec.json [hunk 1]
```diff
@@ -46,11 +46,79 @@
...
         "order": 4
       },
+      "ssl_mode": {
+        "title": "SSL modes",
+        "description": "SSL connection modes. \n <b>disable</b> - Chose this mode to disable encryption of communication between Airbyte and destination database\n <b>allow</b> - Chose this mode to enable encryption only when required by the source database\n <b>prefer</b> - Chose this mode to allow unencrypted connection only if the source database does not support encryption\n <b>require</b> - Chose this mode to always require encryption. If the source database server does not support encryption, connection will fail  \n See more information - <a href=\"https://jdbc.postgresql.org/documentation/head/ssl-client.html\"> in the docs</a>.",
+        "type": "object",
+        "order": 5,
+        "oneOf": [
+          {
+            "title": "disable",
```
Comment (sajarin) [inline airbyte-integrations/connectors/destination-clickhouse-strict-encrypt/src/test/resources/expected_spec.json:56]: Seems like according to the clickhouse documentation, the possible values for the ssl_mode are "none" or "strict" or "relaxed", we probably only need to keep "none and "strict" here to keep the same functionality. 

Also slight typo "Chose" should be "Choose" here :)

File: airbyte-integrations/connectors/destination-clickhouse/src/main/resources/spec.json [hunk 2]
```diff
@@ -46,18 +46,83 @@
...
-        "title": "JDBC URL Params",
-        "type": "string",
-        "order": 5
+      "ssl_mode": {
+        "title": "SSL modes",
+        "description": "SSL connection modes. \n <b>disable</b> - Chose this mode to disable encryption of communication between Airbyte and destination database\n <b>allow</b> - Chose this mode to enable encryption only when required by the source database\n <b>prefer</b> - Chose this mode to allow unencrypted connection only if the source database does not support encryption\n <b>require</b> - Chose this mode to always require encryption. If the source database server does not support encryption, connection will fail\n  <b>verify-ca</b> - Chose this mode to always require encryption and to verify that the source database server has a valid SSL certificate\n  <b>verify-full</b> - This is the most secure mode. Chose this mode to always require encryption and to verify the identity of the source database server\n See more information - <a href=\"https://jdbc.postgresql.org/documentation/head/ssl-client.html\"> in the docs</a>.",
+        "type": "object",
+        "order": 7,
+        "oneOf": [
+          {
```
Comment (sajarin) [inline airbyte-integrations/connectors/destination-clickhouse/src/main/resources/spec.json:55]: See comments above about the supported ssl modes and the slight typo.

**Review by sajarin — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-clickhouse/src/main/resources/spec.json (hunk 2 above)
Comment (StellaMaUCI) [reply airbyte-integrations/connectors/destination-clickhouse/src/main/resources/spec.json:55]: Sure! I will change it soon

**Pull request closed without merging**